crates/log-server/src/loglet_worker.rs

```diff
@@ -422,6 +422,9 @@ impl<S: LogStore> LogletWorker<S> {
             // If shutdown happened, this task will be disposed of and we won't send
             // the response.
             match msg.query {
+                TailUpdateQuery::Unknown => {
+                    unreachable!()
+                }
```

Comment:
Wouldn't it be better to fail the operation and return TailUpdated with_status(Status::Malformed)?

Reply:
Oh, definitely! Sorry I didn't see it.